[ECOINT-40] Add Gravitee integration (#2695)
DataDog/integrations-extras: gravitee/README.md

# Gravitee APIM

## Overview

Gravitee APIM is an enterprise-class agentic AI, API, and event stream management platform. It enables you to manage event streams and agentic AI as securely and easily as traditional APIs, bringing all your APIs together in a unified, full lifecycle management solution.

This integration allows you to monitor API and event stream performance, connection details, and API Gateway metrics, providing insights into the health and behavior of your Gravitee infrastructure. You can track top APIs, request and response times, content sizes, latency, slow performing APIs, gateway performance metrics, logs, and more.

This integration includes a pre-built Datadog dashboard to support comprehensive monitoring and observability of Gravitee APIM.

## Setup

> ### Download and Install

Reviewer: ```suggestion
### Download and install
```

# Gravitee

## Overview

> Comprehensive monitoring and observability of the Gravitee API & Event Streams Management solution. This integration allows you to track API & Event Stream performance, connection details, and database metrics, providing insights into the health and behavior of your Gravitee infrastructure.

Reviewer: The first sentence is a fragment, not a full sentence. Can you reword it into a full sentence?

> We rely on your feedback and feature requests to improve this dashboard. Please send feedback to <contact@graviteesource.com>.

Reviewer: I'm unclear about which dashboard this line refers to. Also, using "we" can be ambiguous, since it's not clear whether "we" means Gravitee or Datadog. Can you reword that part?

## Setup

## Download and install

> To configure the Datadog reporter, download the reporter plugin [here][1]. Once you've downloaded the .ZIP file, you can add it to the Gateway in the same way as [other plugins][2]. Typically, you'll install plugins in the `/plugins` directory of your installation. As with other reporters, the Datadog reporter plugin only needs to be installed on the Gateway, not the Management API.

Reviewer: ```suggestion
To configure the Datadog reporter, download the [reporter plugin][1]. After you download the .ZIP file, you can add it to the Gateway in the same way as [other plugins][2]. Typically, you install plugins in the `/plugins` directory of your installation. As with other reporters, the Datadog reporter plugin only needs to be installed on the Gateway, not the Management API.
```

{% hint style="info" %}

> If you want to collect system metrics and logs from the Management API service, use the [Datadog agent][3] to tail the Management API logs, or collect them from stdout.

Reviewer: ```suggestion
If you want to collect system metrics and logs from the Management API service, use the [Datadog Agent][3] to tail the Management API logs, or collect them from stdout.
```

{% endhint %}

If you are installing the Gravitee Gateway via Helm, add the following entry in the `additionalPlugins` section (changing the version as needed):

```yaml
gateway:
  additionalPlugins:
    - https://download.gravitee.io/graviteeio-ee/apim/plugins/reporters/gravitee-reporter-datadog/gravitee-reporter-datadog-3.0.1.zip
```

## Configuration

> To configure the Datadog reporter on the Gateway, enable the `reporters` section in `gravitee.yml`. This will look something like:

Reviewer: ```suggestion
To configure the Datadog reporter on the Gateway, enable the `reporters` section in `gravitee.yml`. This looks something like:
```

```yaml
reporters:
  datadog:
    enabled: true
    site: "datadoghq.eu"
    authentication:
      #apiKeyPrefix: ""
      apiKey: "YOUR_API_KEY"
      #appKey: "YOUR_APP_KEY"
      #tokenScheme: ""
      #token: "YOUR_TOKEN"
      #username: "YOUR_USERNAME"
      #password: "YOUR_PASSWORD"
```

> Authentication is required for the Gateway to send reporting data to Datadog. Gravitee sends data to Datadog as an [API client][4] over HTTP, and so needs to authenticate to Datadog. The basic way to do this is via an [API key][5], but you can also configure application keys and client tokens, depending on what your Datadog account requires.

Reviewer: ```suggestion
Authentication is required for the Gateway to send reporting data to Datadog. Gravitee sends data to Datadog as an [API client][4] over HTTP, and therefore needs to authenticate to Datadog. The basic way to do this is through an [API key][5], but you can also configure application keys and client tokens, depending on what your Datadog account requires.
```

Gravitee has different types of reporting data, and each type maps to a different resource type in Datadog. The mapping is as follows:

| Gravitee Convention                      | Examples                      | Datadog Convention |
| ---------------------------------------- | ----------------------------- | ------------------ |
| Metadata                                 | API name, user agent          | Tags               |
| Monitoring                               | CPU, memory usage             | Metrics            |
| EndpointStatus                           | Health check status           | Events             |
| [Metrics][7] | Response time, content length | Metrics            |
| [Logs][8]   | Request body, response body   | Log                |

The reporter sends metrics to Datadog with the prefix `gravitee.apim`. Metrics in Datadog appear with underscores between words, instead of the CamelCase default shown in the metrics page. For example, `proxyResponseTimeMs` appears in Datadog as `proxy_response_time_ms`.

For details on what metrics and logs are being sent, please see [documentation][9] and [plugin details][10].

## Uninstallation

## Uninstall

> Simply disable or remove the DataDog Reporter plugin in your Gravitee values.yaml file.

Reviewer: ```suggestion
Disable or remove the DataDog Reporter plugin in your Gravitee values.yaml file.
```
